src/Microsoft.Health.SqlServer/Features/Schema/SchemaInitializer.cs

```diff
@@ -18,33 +20,37 @@ namespace Microsoft.Health.SqlServer.Features.Schema
     /// </summary>
     public class SchemaInitializer : IStartable
     {
+        private const string MasterDatabase = "master";
         private readonly SqlServerDataStoreConfiguration _sqlServerDataStoreConfiguration;
         private readonly SchemaUpgradeRunner _schemaUpgradeRunner;
         private readonly SchemaInformation _schemaInformation;
         private readonly ILogger<SchemaInitializer> _logger;
+        private readonly ISqlConnectionFactory _sqlConnectionFactory;
         private bool _started;
 
-        public SchemaInitializer(SqlServerDataStoreConfiguration sqlServerDataStoreConfiguration, SchemaUpgradeRunner schemaUpgradeRunner, SchemaInformation schemaInformation, ILogger<SchemaInitializer> logger)
+        public SchemaInitializer(SqlServerDataStoreConfiguration sqlServerDataStoreConfiguration, SchemaUpgradeRunner schemaUpgradeRunner, SchemaInformation schemaInformation, ISqlConnectionFactory sqlConnectionFactory, ILogger<SchemaInitializer> logger)
         {
             EnsureArg.IsNotNull(sqlServerDataStoreConfiguration, nameof(sqlServerDataStoreConfiguration));
             EnsureArg.IsNotNull(schemaUpgradeRunner, nameof(schemaUpgradeRunner));
             EnsureArg.IsNotNull(schemaInformation, nameof(schemaInformation));
+            EnsureArg.IsNotNull(sqlConnectionFactory, nameof(sqlConnectionFactory));
             EnsureArg.IsNotNull(logger, nameof(logger));
 
             _sqlServerDataStoreConfiguration = sqlServerDataStoreConfiguration;
             _schemaUpgradeRunner = schemaUpgradeRunner;
             _schemaInformation = schemaInformation;
+            _sqlConnectionFactory = sqlConnectionFactory;
             _logger = logger;
         }
 
-        public void Initialize(bool forceIncrementalSchemaUpgrade = false)
+        public async void InitializeAsync(bool forceIncrementalSchemaUpgrade = false)
```

Comment:
here too: avoid async void

Reply:
Not sure if this can be changed as this is called in Start method which is basically an implementation of IStartable. Making Invoke method returning Task would require making Start method async which would require changes in IStartable, ultimately requiring changes in all the places where it is implemented.